### PR TITLE
[FIX] Polynomial Classification: many bugs

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Several errors.

##### Description of changes
1) Disable tests which take too much time (Travis failed due to `TimeoutError`).
2) Upgrade to Inputs and Outputs and fix tests in all four widgets.
3) Upgrade to `class Warning`.
4) Upgrade to iteration over `domain.variables` and not `domain`.
3) Concatenate lists not tuple and lists in **Polynomial Classification**.
4) Fitting bug in **Polynomial Classification** handled.

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3-educational/issues/397068360/

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3-educational/issues/323251041/

**Comment by kernc** [comment]
lmk when this is ready to review.

File: orangecontrib/educational/widgets/owgradientdescent.py [hunk 1]
```diff
@@ -155,10 +155,13 @@ class OWGradientDescent(OWWidget):
...
-    outputs = [("Model", Model),
-               ("Coefficients", Table),
-               ("Data", Table)]
+    class Inputs:
+        data = Input("Data", Table)
+
+    class Outputs:
+        model = Output("Model", Model)
+        coefficients = Output("Coefficients", Table)
+        data = Output("Data", Table, default=True)
```
Comment (kernc) [inline orangecontrib/educational/widgets/owgradientdescent.py]: Why is Data new default?

File: orangecontrib/educational/widgets/tests/test_owpolynomialclassification.py [hunk 2]
```diff
@@ -446,104 +447,121 @@ def test_update_model(self):
...
+        GH-38
+        """
+        w = self.widget
+
+        self.assertFalse(w.Error.fitting_failed.is_shown())
+        learner = LogisticRegressionLearner()
+        learner.preprocessors = [Discretize()]
+        self.send_signal(w.Inputs.learner, learner)
+        self.send_signal(w.Inputs.data, self.iris)
+        self.assertTrue(w.Error.fitting_failed.is_shown())
```
Comment (kernc) [inline orangecontrib/educational/widgets/tests/test_owpolynomialclassification.py]: Why does this fail?

File: orangecontrib/educational/widgets/owgradientdescent.py (hunk 1 above)
Comment (jerneju) [reply orangecontrib/educational/widgets/owgradientdescent.py]: Fixed.

File: orangecontrib/educational/widgets/tests/test_owpolynomialclassification.py (hunk 2 above)
Comment (kernc) [reply orangecontrib/educational/widgets/tests/test_owpolynomialclassification.py]: @jerneju, @PrimozGodec Why does this fail?

Comment (jerneju) [reply orangecontrib/educational/widgets/tests/test_owpolynomialclassification.py]: Classification requires at least two numerical features.

**Pull request merged**